### DISCUSSION TITLE
MAUI Firebase iOS push not working

### DISCUSSION THREAD

**Discussion by Yahya0007** [shinyorg/shiny#1377] [Q&A]
Hi

Android side is working and notification is being received in OnReceived. 

However, iOS side the token is created in CheckPermission(). Howver when test message sent from Firebase console OnReceived is not triggered. 

Is the testing possible on a physical iOS device or the only way to test is via TestFlight or AdHoc distribution?

Thanks

Regards

**Comment by aritchie** [comment]
You have to provide more details than "it's not working please".  A reproducible sample of some sort.  For the rest of your question, please consult the Google firebase documentation

**Reply by Yahya0007** [reply]
Hi Allan

Many thanks for your help. I have rephrased the original question and am also going through Firebase docs in details. I’ll give it another go and if nothing works then will come back with a sample.

Thanks again.

Regards

**Reply by aritchie** [reply]
You have to send "content-available: 1" in your iOS payload which is covered in the shiny docs

**Reply by Yahya0007** [reply]
In CheckPermission(), for iOS the token is of type; E1D93FA221C8518CDD0855E06382E8676111A467C21C5014F49C1E7A37655DCE 

This token when used in client app comes up with 'The registration token is not a valid FCM registration token'.

I am not sure why the token is of wrong type. GoogleService-Info.plist is in the root folder of the MAUI project and registered as;

	<ItemGroup>
		<BundleResource Include="GoogleService-Info.plist" />
	</ItemGroup>

In Android the correct token of type 'eo-Jtpx6RdOX_xuGzfqJnY:AP...' is received and sending push notification from client with this token works fine. So issue is on the iOS side only.

**Reply by aritchie** [reply]
Please submit a reproducible sample using the proper issue template!

**Reply by Yahya0007** [reply]
This is now done under issue #1382. Many thanks for all your help.
